### PR TITLE
docs(aio): document Custom Async validators

### PR DESCRIPTION
**Pull request by karptonite**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

I spent a while debugging an Async validator that emitted the correct result but never completed. I hope that this addition to the docs will save other developers from falling into that trap. 

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: aio/content/guide/form-validation.md [hunk 1]
```diff
@@ -139,6 +139,10 @@ null if the control value is valid _or_ a validation error object.
 The validation error object typically has a property whose name is the validation key, `'forbiddenName'`,
 and whose value is an arbitrary dictionary of values that you could insert into an error message, `{name}`.
 
+Custom Async validators are similar to Sync validators, but they must instead return a Promise or Observable
```
Comment (kapunahelewong) [inline aio/content/guide/form-validation.md]: Could you make these lowercase:

Async --> async 
Sync --> sync

File: aio/content/guide/form-validation.md [hunk 2]
```diff
@@ -139,6 +139,10 @@ null if the control value is valid _or_ a validation error object.
...
+that later emits null or a validation error object. In the case of an Observable, the Observable must complete,
+and the form will use the last value emitted for validation.
+
```
Comment (kapunahelewong) [inline aio/content/guide/form-validation.md]: How about this?: 

"In the case of an Observable, the Observable must complete, at which point the form uses the last value emitted for validation."

Does that still carry your intent?

**Review by kapunahelewong — CHANGES_REQUESTED** [review]
@karptonite Thanks for this. I have a couple of editorial suggestions. Let me know your thoughts.

**Comment by karptonite** [comment]
@kapunahelewong Sounds good--I've incorporated your suggestions to the pull request.

**Review by kapunahelewong — APPROVED** [review]

**Comment by kapunahelewong** [comment]
@karptonite Looks great. Would you mind squashing? :)

**Comment by karptonite** [comment]
@kapunahelewong squashed.

**Comment by kapunahelewong** [comment]
@juleskremer This is ready for your review. Thank you!

**Review by juleskremer — APPROVED** [review]

**Pull request merged**